src/MissionEditor/MissionEditor.qml

```diff
@@ -498,6 +500,11 @@ QGCView {
                                 controller.removeMissionItem(index)
                             }
 
+                            onInsert: {
+                                var sequenceNumber = controller.insertSimpleMissionItem(editorMap.center, i)
+                                setCurrentItem(sequenceNumber)
+                            }
```

Comment:
Was the insert command just lost along the way somewhere? Strange. While you're in here, could you change the signal for insert to have a more meaningful parameter than "i". Something like insertAfterIndex or other.